### PR TITLE
refactor(flow): Use querySelectorAll to query panels.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #4817

## Summary

refactor(flow): Use querySelectorAll to query panels.

### CODE REVIEW

**Review by jcfranco — APPROVED** [review]
The changelog appears to be outdated, I don't see

> flow: Do not allow nested panels when allowDescendantPanels is true (https://github.com/Esri/calcite-components/issues/4872) ([840b2f1](https://github.com/Esri/calcite-components/commit/840b2f1a7e522ee515c8d4c64bbd9707d61f0538))

Should it be updated to indicate the latest behavior?

**Comment by driskull** [comment]
I'll remove changelog edits. We can fix it as necessary.

File: src/components/flow/flow.tsx [hunk 1]
```diff
@@ -120,33 +98,13 @@ export class Flow {
...
   handleMutationObserverChange = (): void => {
-    const { el, allowDescendantPanels } = this;
-
-    if (!allowDescendantPanels) {
-      return;
-    }
-
     const newPanels: HTMLCalcitePanelElement[] = Array.from(
-      el.querySelectorAll("calcite-panel:not(calcite-panel calcite-panel)")
+      this.el.querySelectorAll("calcite-panel:not(calcite-panel calcite-panel)")
```
Comment (eriklharper) [inline src/components/flow/flow.tsx:102]: Will this query still work if you have a 3rd nested `calcite-panel` like `(calcite-panel calcite-panel calcite-panel)` or even beyond that?

Comment (driskull) [reply src/components/flow/flow.tsx:102]: Wouldn't that fall under `:not(calcite-panel calcite-panel)`?

File: src/components/flow/flow.stories.ts [hunk 2]
```diff
@@ -38,8 +38,8 @@ const createFlowItemAttributes: (group: string) => Attributes = (group) => {
       value: boolean("menuOpen", false, group)
     },
     {
-      name: "summary",
-      value: text("summary", "Summary", group)
+      name: "description",
+      value: text("description", "Description", group)
```
Comment (driskull) [inline src/components/flow/flow.stories.ts:42]: @anveshmekala I updated the flow story to set the panel to use `description` instead of summary. Any other stories we should check to make sure they aren't using the deprecated props?

File: src/components/flow/flow.tsx [hunk 3]
```diff
@@ -119,44 +97,21 @@ export class Flow {
...
-      return;
-    }
-
-    const newPanels = (event.target as HTMLSlotElement)
-      .assignedElements({
-        flatten: true
-      })
-      .filter((el) => el?.matches("calcite-panel")) as HTMLCalcitePanelElement[];
+      el.querySelectorAll("calcite-panel")
+    ).filter((panel) => !panel.closest("calcite-panel")) as HTMLCalcitePanelElement[];
```
Comment (driskull) [inline src/components/flow/flow.tsx:105]: @jcfranco storybook kept failing when using `el.querySelectorAll("calcite-panel:not(calcite-panel calcite-panel)")`.

I'm guessing this is because its stuck on a version that doesn't support it?

https://developer.mozilla.org/en-US/docs/Web/CSS/:not#browser_compatibility

I'd rather use the single selector if possible but storybook is holding us back here.

Comment (driskull) [reply src/components/flow/flow.tsx:105]: cc @eriklharper

File: src/components/flow/flow.tsx [hunk 4]
```diff
@@ -119,44 +97,21 @@ export class Flow {
...
-      return;
-    }
-
-    const newPanels = (event.target as HTMLSlotElement)
-      .assignedElements({
-        flatten: true
-      })
-      .filter((el) => el?.matches("calcite-panel")) as HTMLCalcitePanelElement[];
+      el.querySelectorAll("calcite-panel")
+    ).filter((panel) => !panel.matches("calcite-panel calcite-panel")) as HTMLCalcitePanelElement[];
```
Comment (driskull) [inline src/components/flow/flow.tsx:105]: https://github.com/Esri/calcite-components/pull/4875#discussion_r917120040

File: src/components/flow/flow.tsx (hunk 3 above)
Comment (jcfranco) [reply src/components/flow/flow.tsx]: > I'd rather use the single selector if possible but storybook is holding us back here.

Yeah, that'd be cleaner. For now, let's proceed with this tested version and we should create a refactor issue to make the switch once newer browsers are used for screenshot tests.

File: src/components/flow/flow.stories.ts (hunk 2 above)
Comment (jcfranco) [reply src/components/flow/flow.stories.ts:42]: This should be in a separate PR as it's unrelated to the overall changes.

File: src/components/flow/flow.tsx [hunk 5]
```diff
@@ -119,44 +97,21 @@ export class Flow {
...
-    const { panels } = this;
     const oldPanelCount = panels.length;
     const newPanelCount = newPanels.length;
     const activePanel = newPanels[newPanelCount - 1];
     const previousPanel = newPanels[newPanelCount - 2];
 
     if (newPanelCount && activePanel) {
       newPanels.forEach((panelNode) => {
-        panelNode.showBackButton = newPanelCount > 1;
+        panelNode.showBackButton = panelNode === activePanel && newPanelCount > 1;
```
Comment (jcfranco) [inline src/components/flow/flow.tsx:114]: Is this a separate fix? It seems unrelated to the panel query behavior.

**Review by jcfranco — APPROVED** [review]
💥

File: src/components/flow/flow.stories.ts (hunk 2 above)
Comment (driskull) [reply src/components/flow/flow.stories.ts:42]: Ok. I'll revert.

File: src/components/flow/flow.tsx (hunk 5 above)
Comment (driskull) [reply src/components/flow/flow.tsx:114]: yeah ill separate it out.

File: src/components/flow/flow.tsx (hunk 3 above)
Comment (eriklharper) [reply src/components/flow/flow.tsx]: Yeah, unfortunately we'll have to just deal with the outdated Chrome for now until we can make the switch to Chromatic.
